### PR TITLE
SUS-1191: Copy single filter content into bulk mode text box when switching modes

### PR DESCRIPTION
**Pull request by mszabo-wikia**
ticket & QA notes: https://wikia-inc.atlassian.net/browse/SUS-1191

### CODE REVIEW

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 1]
```diff
@@ -45,7 +50,7 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
 
 		// handle "validate regex" button
 		on('click', '#validate', function(ev) {
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:59]: MAJOR Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 2]
```diff
@@ -2,6 +2,9 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
 	// edit token is required by Phalanx API
 	phalanx.init(mw.config.get('wgPhalanxToken'));
 
+	var $filter = $('#wpPhalanxFilter'),
+		$bulkFilter = $('#wpPhalanxFilterBulk');
+
 	$('body').
 		// handle blocks "unblocking" (i.e. removing blocks)
 		on('click', 'button.unblock', function(ev) {
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:10]: MAJOR Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [SpecialPhalanx.js#L32](https://github.com/Wikia/app/blob/fe792456c2554c919cb23535e4a7f2afc3d56f2a/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L32): Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [SpecialPhalanx.js#L43](https://github.com/Wikia/app/blob/fe792456c2554c919cb23535e4a7f2afc3d56f2a/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L43): Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [SpecialPhalanx.js#L73](https://github.com/Wikia/app/blob/fe792456c2554c919cb23535e4a7f2afc3d56f2a/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L73): Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
- MAJOR 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [SpecialPhalanx.js#L32](https://github.com/Wikia/app/blob/be07be175c096bda81264df15577a2f5268d1e4c/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L32): Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
2. MAJOR [SpecialPhalanx.js#L47](https://github.com/Wikia/app/blob/be07be175c096bda81264df15577a2f5268d1e4c/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L47): Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
3. MAJOR [SpecialPhalanx.js#L80](https://github.com/Wikia/app/blob/be07be175c096bda81264df15577a2f5268d1e4c/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L80): Remove the unused function parameter "ev". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 3]
```diff
@@ -30,8 +33,14 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
...
 				bulkModeWrapper = $('#bulkmode');
 
+			// SUS-1191: preload single filter contents into bulk mode textbox
+			$bulkFilter.val($filter.val());
+
 			singleModeWrapper.slideUp();
 			bulkModeWrapper.slideDown();
+
+			// SUS-1191: autofocus newly revealed input field
+			$bulkFilter.focus();
```
Comment (macbre) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:43]: :+1:

**Review by macbre — APPROVED** [review]

**Pull request merged**